packages/staking/src/features/activity/RewardsHistory.tsx

```diff
@@ -10,18 +12,34 @@ type RewardsHistoryProps = {
   groupedRewardsActivities: AssetActivityListProps[];
   walletActivitiesStatus: StateStatus;
 };
+
+const FAKE_LOAD_TIMEOUT = 1000;
+
 export const RewardsHistory = ({ groupedRewardsActivities, walletActivitiesStatus }: RewardsHistoryProps) => {
   const { t } = useTranslation();
 
+  const pageSize = useGroupedActivitiesPageSize();
+  const [paginatedLists, setPaginatedLists] = useState<AssetActivityListProps[]>([]);
+
+  const loadMoreData = useCallback(async () => {
+    if (groupedRewardsActivities.length === 0) return;
+    setTimeout(() => {
+      setPaginatedLists((prevList) => take(groupedRewardsActivities, prevList.length + pageSize));
+    }, FAKE_LOAD_TIMEOUT);
```

Comment:
Can we remove this to load faster?
```suggestion
      setPaginatedLists((prevList) => take(groupedRewardsActivities, prevList.length + pageSize));
```